actor/process_test.go

```diff
@@ -48,3 +48,56 @@ func Test_CleanTrace(t *testing.T) {
 func panicWrapper() {
 	panic("foo")
 }
+
+func Test_StartupMessages(t *testing.T) {
+	e, err := NewEngine(NewEngineConfig())
+	require.NoError(t, err)
+
+	type testMsg struct{}
+
+	msgCh := make(chan any, 10) // used to check the msg order
+
+	go func() {
+		e.SpawnFunc(func(c *Context) {
+			fmt.Printf("Got message type %T\n", c.Message())
+			switch c.Message().(type) {
+			case Initialized:
+				time.Sleep(10 * time.Millisecond) // sleep to simulate db load
+			case Started:
+				time.Sleep(10 * time.Millisecond) // sleep to sumulate db load
+			}
+			msgCh <- c.Message()
+		}, "foo", WithID("bar"))
+	}()
+
+	time.Sleep(5 * time.Millisecond)
```

Comment:
I understand why this is needed, but 5 milliseconds is quite large. I wonder if we have no other mechanism to wait for things in tests, so that we do not need to relay on `time.Sleeps` all the time.

Reply:
Goo point. I just made a commit which uses channels for the synchronization instead of the sleeps. Thanks!